### PR TITLE
ESDT NFT Marketplace documentation

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: contracts/examples/esdt-nft-marketplace/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Difference between ESDT NFTs and ERC1155 NFTs
+
+Before we start with the marketplace, there's something else that needs to be covered. If you're familiar with Ethereum, you probably heard about the ERC1155 standard for NFTs. There are a few problems with that standard.  
+
+The main problem is that it is very centralized. A single smart contract holds the data for all NFTs, and all the related operations, like transfers, are also done through that contract. This can lead to trust issues or bottlenecks.  
+
+Another problem is that using said NFTs as payment for smart contracts is very difficult. It requires a call to the main ERC1155 contract, which then calls the receiver smart contract, which has to implement a very specific method for receiving the token. As you can probably guess, this adds a lot of overhead and costs a ton of gas. And now the receiver smart contract owner has another problem: how to get that NFT to its own user account? This requires yet another call to the ERC1155 contract, through the receiver contract, which is another two SC calls. More overhead, more gas.
```
Comment (andrei-marinica) [inline contracts/examples/esdt-nft-marketplace/README.md:17]: a lot of gas (ton = too informal)

Comment (dorin-iancu) [reply contracts/examples/esdt-nft-marketplace/README.md]: Fixed.

File: contracts/examples/esdt-nft-marketplace/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+```
+#[endpoint(endAuction)]
+fn end_auction(&self, nft_type: TokenIdentifier, nft_nonce: u64)
+```
+
+Arguments are the same as the `bid` ones. If no bids were made, the NFT is returned to the owner. If bids were made, the NFT is sent to the highest bidder, and the bid is split between the NFT creator, marketplace SC and NFT owner.  
+
+## Conclusion
+
+NFTs are the future of blockchain, and can be used both for entertainment purposes (like cryptokitties, for example) and for serious purposes (like ownership for art, music, etc.). In both cases, the marketplace can be used to sell NFTs as easily as possible.
```
Comment (andrei-marinica) [inline contracts/examples/esdt-nft-marketplace/README.md:104]: ownership of art

File: contracts/examples/esdt-nft-marketplace/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+By implementing NFTs on the protocol level, we've solved those problems by treating NFTs as ESDTs with additional metadata. This means your user account can *own* the NFT natively instead of relying on an external contract. You can also transfer that NFT at will, and even use it as payment for smart contracts. And what's more, said contract doesn't even need to implement some standardized interface!  
+
+An ESDT NFT has a token identifier (like all the rest), which is also referred to as its "type" or "class". Then, a type can have "subtypes" or "subclasses", differentiated by `nonce`, `name`, and additional user-defined `attributes`.  
+
+Now that we've hyped ESDT NFTs, it's time to see how you can sell them through the marketplace.  
+
+## The Marketplace
+
+The marketplace has been designed to be as simple as possible. All it does is accept an NFT and auction it with your terms. It accepts bids from other accounts, and concludes the auction by either sending the token back to you if no bids were received, or send you the respective bid amount and the token to the new owner.
```
Comment (andrei-marinica) [inline contracts/examples/esdt-nft-marketplace/README.md:27]: I don't think "concludes"  can be used like that, use "finishes" instead

better: ... and finishes the auction by either sending the token to the winner, or back to the owner if no bids were received.

File: contracts/examples/esdt-nft-marketplace/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+By implementing NFTs on the protocol level, we've solved those problems by treating NFTs as ESDTs with additional metadata. This means your user account can *own* the NFT natively instead of relying on an external contract. You can also transfer that NFT at will, and even use it as payment for smart contracts. And what's more, said contract doesn't even need to implement some standardized interface!  
+
+An ESDT NFT has a token identifier (like all the rest), which is also referred to as its "type" or "class". Then, a type can have "subtypes" or "subclasses", differentiated by `nonce`, `name`, and additional user-defined `attributes`.  
+
+Now that we've hyped ESDT NFTs, it's time to see how you can sell them through the marketplace.
```
Comment (andrei-marinica) [inline contracts/examples/esdt-nft-marketplace/README.md:23]: Now that we have introduced NFT, it is time to see how they can be sold on the marketplace.

File: contracts/examples/esdt-nft-marketplace/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
+# ESDT NFT Marketplace
+
+## Introduction
+
+Whether you've been familiar with the world of cryptocurrency and blockchains for a while or you're just starting out, you've most likely heard of NFTs (which stands for Non Fungible Token). To put it simply, an NFT is used to represent a unique item, by providing something known as "digital scarcity". NFTs could be used to represent collectibles, contracts, and much more.  
+
+As we all know, there really isn't such thing as "unique data" when it comes to computers. They copy and share data all the time. The key difference here is the immutability of data on the blockchain: Once an NFT is created, you will be able to trace its history through time, without having to worry about a third party modifying, or even deleting that said NFT.  
+
+But what if you decide you don't really have any value for said collectible item? That's where the NFT marketplace comes in.
```
Comment (andrei-marinica) [inline contracts/examples/esdt-nft-marketplace/README.md:9]: But what if someone wants to exchange said collectible item for something else?

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**